core/service/src/vault/storage/crypto_material/base.rs

```diff
@@ -668,12 +673,17 @@ where
                     e
                 );
             }
-            let vault_result_is_err = match back_vault {
+
+            priv_result.is_err()
+        };
+        let f3 = async {
+            // TODO should we indeed delete backup material here?
```

Comment:
WDYT reviewer? Should a backup of a CRS or key also be deleted when we delete the CRS or key?